runtime/src/drivers.rs

```diff
@@ -0,0 +1,210 @@
+// Licensed under the Apache-2.0 license
+
+#![cfg_attr(not(feature = "fip-self-test"), allow(unused))]
+
+#[cfg(feature = "fips_self_test")]
+pub use crate::fips::{fips_self_test_cmd, fips_self_test_cmd::SelfTestStatus};
+
+use crate::{
+    dice, CptraDpeTypes, DpeCrypto, DpePlatform, Mailbox, DPE_SUPPORT, MAX_CERT_CHAIN_SIZE,
+};
+
+use arrayvec::ArrayVec;
+use caliptra_drivers::{
+    CaliptraError, CaliptraResult, DataVault, Ecc384, KeyVault, Lms, PersistentDataAccessor,
+    ResetReason, Sha1, SocIfc,
+};
+use caliptra_drivers::{Hmac384, PcrBank, PcrId, Sha256, Sha384, Sha384Acc, Trng};
+use caliptra_registers::mbox::enums::MboxStatusE;
+use caliptra_registers::{
+    csrng::CsrngReg, dv::DvReg, ecc::EccReg, entropy_src::EntropySrcReg, hmac::HmacReg, kv::KvReg,
+    mbox::MboxCsr, pv::PvReg, sha256::Sha256Reg, sha512::Sha512Reg, sha512_acc::Sha512AccCsr,
+    soc_ifc::SocIfcReg, soc_ifc_trng::SocIfcTrngReg,
+};
+use dpe::{
+    commands::{CommandExecution, DeriveChildCmd, DeriveChildFlags},
+    context::ContextHandle,
+    dpe_instance::{DpeEnv, DpeInstance, DpeTypes},
+    support::Support,
+    DPE_PROFILE,
+};
+
+use crypto::{AlgLen, Crypto};
+
+pub struct Drivers {
+    pub mbox: Mailbox,
+    pub sha_acc: Sha512AccCsr,
+    pub data_vault: DataVault,
+    pub key_vault: KeyVault,
+    pub soc_ifc: SocIfc,
+    pub sha256: Sha256,
+
+    // SHA2-384 Engine
+    pub sha384: Sha384,
+
+    // SHA2-384 Accelerator
+    pub sha384_acc: Sha384Acc,
+
+    /// Hmac384 Engine
+    pub hmac384: Hmac384,
+
+    /// Cryptographically Secure Random Number Generator
+    pub trng: Trng,
+
+    /// Ecc384 Engine
+    pub ecc384: Ecc384,
+
+    pub persistent_data: PersistentDataAccessor,
+
+    pub lms: Lms,
+
+    pub sha1: Sha1,
+
+    pub pcr_bank: PcrBank,
+
+    pub cert_chain: ArrayVec<u8, MAX_CERT_CHAIN_SIZE>,
+
+    pub attestation_disabled: bool,
+
+    #[cfg(feature = "fips_self_test")]
+    pub self_test_status: SelfTestStatus,
+
+    pub is_shutdown: bool,
```

Comment:
This will be cleared on warm/update reset. Is that expected? Otherwise we should maybe put this in the FHT.

Reply:
We don't make the shutdown condition persistent.